src/charm.py

```diff
@@ -203,6 +211,29 @@ def _generate_webui_config_file(self) -> str:
             auth_database_url=self._get_auth_database_url(),
         )
 
+    def _on_sdcore_config_relation_joined(self, event: RelationJoinedEvent) -> None:
+        """Handle sdcore_config relation joined event.
+
+        Set the Webui config URL in the relation databag when Webui becomes accessible.
+
+        Args:
+            event: RelationJoinedEvent
+        """
+        if not self._webui_service_is_running():
+            return
+
+        webui_config_url = self._get_webui_config_url()
+        self._sdcore_config.set_webui_url(
+            webui_url=webui_config_url,
+            relation_id=event.relation.id,
+        )
```

Comment:
Do we need this event at all? We are already doing this in the `_configure` method. 

We can simply replace 
```
self.framework.observe(
            self.on.sdcore_config_relation_joined, self._on_sdcore_config_relation_joined
        )
```

with 

```
self.framework.observe(
            self.on.sdcore_config_relation_joined, self._configure
        )
```

and get rid of this event handler

Reply:
Done.